src/Tools/dotnet-monitor/Auth/GeneratedJwtKey.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal sealed class GeneratedJwtKey
+    {
+        public readonly string Token;
+        public readonly string Subject;
+        public readonly string PublicKey;
+
+        private GeneratedJwtKey(string token, string subject, string publicKey)
+        {
+            this.Token = token;
+            this.Subject = subject;
+            this.PublicKey = publicKey;
+        }
+
+        public static GeneratedJwtKey Create()
+        {
+            Guid subjectId = Guid.NewGuid();
+            string subjectStr = subjectId.ToString("D");
+
+            ECDsa dsa = ECDsa.Create();
+            dsa.GenerateKey(ECCurve.NamedCurves.nistP384);
+            ECDsaSecurityKey secKey = new ECDsaSecurityKey(dsa);
+            SigningCredentials signingCreds = new SigningCredentials(secKey, SecurityAlgorithms.EcdsaSha384);
+            JwtHeader newHeader = new JwtHeader(signingCreds, null, JwtConstants.HeaderType);
+
+            Claim issClaim = new Claim("iss", AuthConstants.ApiKeyJwtIssuer);
+            Claim subClaim = new Claim("sub", subjectStr);
+            JwtPayload newPayload = new JwtPayload(new Claim[] { issClaim, subClaim });
+
+            JwtSecurityToken newToken = new JwtSecurityToken(newHeader, newPayload);
+
+            ECDsa pubDsa = ECDsa.Create(dsa.ExportParameters(includePrivateParameters: false));
+            ECDsaSecurityKey pubSecKey = new ECDsaSecurityKey(pubDsa);
+            JsonWebKey jwk = JsonWebKeyConverter.ConvertFromECDsaSecurityKey(pubSecKey);
+            string publicKeyJson = JsonSerializer.Serialize(jwk, new JsonSerializerOptions() { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull });
```

Comment:
I don't think this compiles for .NET Core 3.1

Reply:
This does not compile with `System.Text.Json` < 5.0. However, dotnet-monitor uses System.Text.Json version 5.0.0 netcoreapp3.0 assembly. This apparently is actually not a breaking change but is a breaking change in an independently versioning assembly.